### PR TITLE
Convert more utils to TypeScript

### LINKED ISSUE #927: Add types to `utils/format-strings.js`
## Description

Add type checking to `format-strings.js`. This module has zero dependencies. Make sure to add it to `tsconfig.json`'s `include` list.

## Additional context

Part of an ongoing effort to add type checking to parts of the project that can be type checked.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### LINKED ISSUE #933: Add types to `utils/math.js`
## Description

Add type checking to `math.js`. This module has zero dependencies and is already type annotated. Add it to `tsconfig.json`'s `include` list and fix any type errors that arise.

## Additional context

Part of an ongoing effort to add type checking to parts of the project that can be type checked.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### LINKED ISSUE #935: Add types to `utils/prng.js`
## Description

Add type checking to `prng.js`. This module has zero dependencies and is already type annotated. Make sure to add it to `tsconfig.json`'s `include` list and correct any potential type errors that arise.

## Additional context

Part of an ongoing effort to add type checking to parts of the project that can be type checked.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### LINKED ISSUE #937: Add types to `utils/send-message.js`
## Description

Add type checking to `send-message.js`. This module has one dependency that will be typed in #924. Make sure to add it to `tsconfig.json`'s `include` list.

Blocked by #924 

## Additional context

Part of an ongoing effort to add type checking to parts of the project that can be type checked.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### LINKED ISSUE #939: Add types to `utils/srand.js`
## Description

Add type checking to `srand.js`. This module has zero dependencies. Make sure to add it to `tsconfig.json`'s `include` list.

## Additional context

Part of an ongoing effort to add type checking to parts of the project that can be type checked.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #927
Fixes #933
Fixes #935
Fixes #937 
Fixes #939

## Description

This PR converts more utils to TypeScript.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/utils/prng.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/**
+/*
```
Comment (sarayourfriend) [inline src/utils/prng.ts:1]: Is this part of the TSDoc? I've never seen a project use single asteriks to open doc strings before... it doesn't render as expected in my editor (maybe I'm alone there though).

Comment (dhruvkb) [reply src/utils/prng.ts:1]: It's weird, the linter doesn't complain whether there's one or two asterisks.

Comment (sarayourfriend) [reply src/utils/prng.ts]: Right, the TSDoc eslint plugin just enforces _TSDoc_ which is a separate thing from JSDoc. I think we'd still want the JSDoc linter in addition to TSDoc, it seems that the TSDoc one just enforces specific tags, not the style of anything. The documentation for the TSDoc plugin is awful too :confused: 

https://tsdoc.org/pages/packages/eslint-plugin-tsdoc/

Or I guess there's just a single rule that enforces all of it? In any case, it leaves out a lot of stuff that the jsdoc plugin covers that would be nice to have still (like the missing asterisks and stuff)

Comment (dhruvkb) [reply src/utils/prng.ts]: Yeah, the TSDoc plugin seems to only validate the syntax and nothing else. The JSDoc plugin is definitely needed.

File: src/components/VFilters/VSearchGridFilter.vue [hunk 2]
```diff
@@ -76,7 +75,7 @@ export default {
       if (filterType === 'searchBy') {
         return ''
       }
-      return i18n.t(`filters.${kebabCase(filterType)}.title`)
+      return i18n.t(`filters.${kebab(filterType)}.title`)
```
Comment (sarayourfriend) [inline src/components/VFilters/VSearchGridFilter.vue:78]: Nice! Love removing code :partying_face:

**Review by sarayourfriend — APPROVED** [review]
This LGTM! Checked it out locally and the types seem good. Glad you found some code to remove too :rocket:

**Review by zackkrida — APPROVED** [review]
lgtm! Nice to only have one pseudo random number generator in the codebase.

**Review by obulat — APPROVED** [review]
LGTM!

File: src/utils/send-message.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { isProd } from '~/utils/node-env'
+
+// TODO: set correct targetOrigin of the parent window
+const TARGET_ORIGIN = '*'
```
Comment (krysal) [inline src/utils/send-message.ts:4]: Is this planned to be resolved here or in a future PR? 👀

**Review by krysal — COMMENTED** [review]
Looks great! Just a little question.

File: src/utils/send-message.ts (hunk 3 above)
Comment (dhruvkb) [reply src/utils/send-message.ts:4]: It's been that way since the beginning 😆

**Pull request merged**